### PR TITLE
Switch async-storage repository

### PR DESCRIPTION
**Pull request by kidroca**
### Details
The package @react-native-community/async-storage is deprecated from @react-native-community repo
It's moved and maintained as @react-native-async-storage/async-storage
There are some improvements that have happened since then
There are no breaking changes listed

This PR needs to be updated with the correct Onyx hash https://github.com/Expensify/react-native-onyx/pull/85 is merged

### Fixed Issues

Related to #2667

### Tests
Since storage affects everything a general tests for regressions should be carried 

### QA Steps
Since storage affects everything a general tests for regressions should be carried 

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

image

#### Mobile Web

image

#### Desktop

image

#### iOS

image

#### Android

image

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -20,6 +21,7 @@ buildscript {
         classpath("com.google.firebase:firebase-crashlytics-gradle:2.3.0")
         // NOTE: Do not place your application dependencies here; they belong
         // in the individual module build.gradle files
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion"
```
Comment (kidroca) [inline android/build.gradle:24]: The changes to this file are not require, they enable the `next` AsyncStorage version 
https://react-native-async-storage.github.io/async-storage/docs/advanced/next

File: android/gradle.properties [hunk 2]
```diff
@@ -26,6 +26,7 @@ android.enableJetifier=true
 
 # Increase storage capacity (the default is 6 MB)
 AsyncStorage_db_size_in_MB=10
+AsyncStorage_useNextStorage=true
```
Comment (kidroca) [inline android/gradle.properties:29]: The changes to this file are not require, they enable the `next` AsyncStorage version 
https://react-native-async-storage.github.io/async-storage/docs/advanced/next

File: android/settings.gradle [hunk 3]
```diff
@@ -1,8 +1,6 @@
 rootProject.name = 'ExpensifyCash'
 include ':react-native-config'
 project(':react-native-config').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-config/android')
-include ':@react-native-community_async-storage'
-project(':@react-native-community_async-storage').projectDir = new File(rootProject.projectDir, '../node_modules/@react-native-community/async-storage/android')
```
Comment (kidroca) [inline android/settings.gradle:5]: These are no longer needed with the new version

File: package.json [hunk 4]
```diff
@@ -80,7 +80,7 @@
     "react-native-image-picker": "^4.0.3",
     "react-native-keyboard-spacer": "^0.4.1",
     "react-native-modal": "^11.10.0",
-    "react-native-onyx": "git+https://github.com/Expensify/react-native-onyx.git#1e82e592032c6d0ede8e40f08beb6be790d149e8",
+    "react-native-onyx": "git+https://github.com/kidroca/react-native-onyx.git#064748fdc6ceef7c9ffe8c5fa62a30ef234aae2a",
```
Comment (kidroca) [inline package.json:83]: This hash should be updated after the Onyx PR is merged

**Review by kidroca — COMMENTED** [review]
I've enabled the `next` AsyncStorage version 
https://react-native-async-storage.github.io/async-storage/docs/advanced/next

I can revert the changes if you prefer
Can't really see any differences in app speeds with either the "old" async storage, the "new" and the "new + next"

**Comment by tgolen** [comment]
Cool, thanks! I think trying out the `next` storage will be fine to do.

File: package.json [hunk 5]
```diff
@@ -38,7 +38,7 @@
     "@formatjs/intl-numberformat": "^6.2.5",
     "@formatjs/intl-pluralrules": "^4.0.13",
     "@onfido/react-native-sdk": "^1.3.3",
-    "@react-native-community/async-storage": "^1.11.0",
+    "@react-native-async-storage/async-storage": "^1.15.5",
```
Comment (tgolen) [inline package.json:41]: Hm, so why are we installing this lib directly? I'm not sure why the previous lib was referenced here either. AFAIK, the only way we should be interfacing with AsyncStorage is through Onyx.

Comment (tgolen) [reply package.json:41]: I'm not seeing it directly imported into any files, so can this just be removed?

**Review by tgolen — CHANGES_REQUESTED** [review]

File: package.json (hunk 5 above)
Comment (kidroca) [reply package.json:41]: This is related to the comment I've made on peerDependencies on the other PR
We need to have it listed as a project dependency

We also have one place where it's imported - inside __mocks__ 
The e2e tests that use Onyx stub AsyncStorage from E.cash __mocks__

**Comment by kidroca** [comment]
I've put this comment on the wrong place, so I'm posting it again here

I've run the 3 version against E.cash and here is a short summary

Median times for 7 app startups per each AsyncStorage version
1. The app is launched
2. The initial chat is always the same - a conversation with 200+ messages
3. Timings are captured after the app stops loading

### Timing for `Onyx.get`
|            |     1.12.1 |     1.15.5 | 1.15.5 next |
|------------|-----------:|-----------:|------------:|
| avg call time   |   131.55ms |   140.95ms |    134.90ms |
| total call time | 32862.34ms | 35175.71ms |  33552.87ms |
| total calls count | 529 | 529 |  529 |
- note: a lot of the calls are resolved from Onyx cache, the actual hard reads vary from 200 to 1000ms

More data is available here:
https://docs.google.com/spreadsheets/d/1ABHCzs9yZIXmHxMJnDboyyMw8ICjQQ3sxIP_23c7Lws/edit?usp=sharing

**Review by tgolen — DISMISSED** [review]

**Comment by tgolen** [comment]
This is approved, but I am adding a `[HOLD]` to the title so that it's not merged until the hash for react-native-onyx is updated.

**Comment by kidroca** [comment]
Updated `react-native-onyx` hash

**Review by tgolen — APPROVED** [review]

**Pull request merged**